libusb/libusbi.h

```diff
@@ -471,25 +471,24 @@ static inline struct libusb_context *usbi_get_context(struct libusb_context *ctx
 	return ctx;
 }
 
-enum usbi_event_flags {
-	/* The list of event sources has been modified */
-	USBI_EVENT_EVENT_SOURCES_MODIFIED = 1U << 0,
+/* event flags of the ctx structure */
+/* The list of event sources has been modified */
+#define USBI_EVENT_EVENT_SOURCES_MODIFIED  (1U << 0)
 
-	/* The user has interrupted the event handler */
-	USBI_EVENT_USER_INTERRUPT = 1U << 1,
+/* The user has interrupted the event handler */
+#define USBI_EVENT_USER_INTERRUPT          (1U << 1)
 
-	/* A hotplug callback deregistration is pending */
-	USBI_EVENT_HOTPLUG_CB_DEREGISTERED = 1U << 2,
+/* A hotplug callback deregistration is pending */
+#define USBI_EVENT_HOTPLUG_CB_DEREGISTERED (1U << 2)
 
-	/* One or more hotplug messages are pending */
-	USBI_EVENT_HOTPLUG_MSG_PENDING = 1U << 3,
+/* One or more hotplug messages are pending */
+#define USBI_EVENT_HOTPLUG_MSG_PENDING     (1U << 3)
 
-	/* One or more completed transfers are pending */
-	USBI_EVENT_TRANSFER_COMPLETED = 1U << 4,
+/* One or more completed transfers are pending */
+#define USBI_EVENT_TRANSFER_COMPLETED      (1U << 4)
 
-	/* A device is in the process of being closed */
-	USBI_EVENT_DEVICE_CLOSE = 1U << 5,
-};
```

Comment:
This seems a backwards change. enums should be encouraged over #defines, not the other way around.

Better would be to do this: https://github.com/libusb/libusb/pull/1417/commits/6b52b8e1508987e906cfddf5573b703cebc9668c

Reply:
Hmm, I don't think we should bump compiler requirements for something so cosmetic. I understand that the suggestion doesn't strictly require C23 but it wouldn't be tested without.

There are certainly reasons for encouraging enums (especially if they had types) over defines in many cases but I am not sure it applies to us here.

As I said initially, I have mixed feelings about all this because we try everything to please compilers but likely make the code less readable. Maybe the best way is just to keep -Wconversion out :)

Reply:
> Hmm, I don't think we should bump compiler requirements for something so cosmetic.

To be clear, I'm not advocating bumping the minimum compiler requirements. As you see in 6b52b8e1508987e906cfddf5573b703cebc9668c, the feature can be wrapped in a macro like we do already for `LIBUSB_DEPRECATED_FOR`. 

>I understand that the suggestion doesn't strictly require C23 but it wouldn't be tested without.

Well sure, but adding a C23 variant to CI shouldn't be too hard.

> There are certainly reasons for encouraging enums (especially if they had types) over defines in many cases but I am not sure it applies to us here.

I have found it very helpful as documenation, especially where those enumerated values are converted in/out of raw uintX_t struct fields.

> As I said initially, I have mixed feelings about all this because we try everything to please compilers but likely make the code less readable. 

I find this change *more* readable because now I, the reader, know the size and range of the thing:

```diff
-enum libusb_endpoint_direction {
+enum libusb_endpoint_direction LIBUSB_ENUM_SIZE(uint8_t) {
```

>Maybe the best way is just to keep -Wconversion out :)

Indeed I have found -Wconversion something that can find problems, but is often too noisy.  I haven't reviewed all this PR yet, but will...